parser-typechecker/src/Unison/DataDeclaration/Dependencies.hs

```diff
@@ -50,67 +50,76 @@ labeledDeclDependenciesIncludingSelfAndFieldAccessors selfRef decl =
           & maybe Set.empty (Set.map LD.TermReferent)
 
 -- | Generate Referents for all possible field accessors of a Decl.
--- Returns 'Nothing' if typechecking of any accessor fails.
-fieldAccessorRefs :: forall v a. (Var.Var v) => Reference -> DD.DataDeclaration v a -> Maybe (Set Referent)
+--
+-- Returns @Nothing@ if this couldn't be a record because it doesn't contain exactly one constructor, or because the
+-- record contains a field with a higher rank type (and thus fails type inference).
+fieldAccessorRefs :: forall v a. (Var v) => TypeReference -> DD.DataDeclaration v a -> Maybe (Set Referent)
 fieldAccessorRefs declRef dd = do
-  -- This ppe is only used for typechecking errors.
-  let ppe = PPE.empty
-  typ <- case DD.constructors dd of
-    [(_, typ)] -> Just typ
-    _ -> Nothing
+  [(_, typ)] <- Just (DD.constructors dd)
+
   -- This name isn't important, we just need a name to generate field names from.
   -- The field names are thrown away afterwards.
   let typeName = Var.named "Type"
   -- These names are arbitrary and don't show up anywhere.
   let vars :: [v]
-      vars = [Var.freshenId (fromIntegral n) (Var.named "_") | n <- [0 .. Type.arity typ - 1]]
-  hashFieldAccessors ppe typeName vars declRef dd
-    <&> \accs ->
-      Map.elems accs
-        & setOf (folded . _1 . to (Reference.DerivedId >>> Referent.Ref))
+      -- We add `n` to the end of the variable name as a quick fix to #4752, but we suspect there's a more
+      -- fundamental fix to be made somewhere in the term printer to automatically suffix a var name with its
+      -- freshened id if it would be ambiguous otherwise.
+      vars = [Var.freshenId (fromIntegral n) (Var.named ("_" <> tShow n)) | n <- [0 .. Type.arity typ - 1]]
```

Comment:
This just copies over a different fix, including its comment, to another place that it should have been applied, but was missed.